chia/data_layer/data_store.py

```diff
@@ -258,8 +290,15 @@ async def insert_into_data_store_from_file(
                         merkle_blob_queries[root_hash_blob].append(index)
                         found_hashes.add(node_hash)
 
-        if found_hashes != set(missing_hashes):
-            raise Exception("Invalid delta file, cannot find all the required hashes")
+            missing_hashes = [hash for hash in missing_hashes if hash not in found_hashes]
+            if missing_hashes:
+                if current_generation < root.generation:
+                    current_generation += 1
+                else:
+                    raise Exception("Invalid delta file, cannot find all the required hashes")
+
+                await self.add_node_hashes(store_id, current_generation)
+                log.info(f"Missing hashes: added old hashes from generation {current_generation}")
```

Comment:
this is inside the `while missing_hashes:` loop but nothing else is using the `current_generation` so it seems odd to be changing that.